### ISSUE TITLE
Create new Brogue Lite release for 1.10.x

### ISSUE THREAD

**Issue by HomebrewHomunculus** [HomebrewHomunculus/BrogueCE#5]
## Background

Currently, the old repository https://github.com/HomebrewHomunculus/BrogueLite/ is on version 1.9.3. Since then, [BrogueCE has released ](https://github.com/tmewett/BrogueCE/releases)1.10 and 1.10.1.

This new repository [was created](https://github.com/HomebrewHomunculus/BrogueCE/pull/2) for easier inclusion of upstream changes into Brogue Lite. It currently includes the changes from the 1.10 version of BrogueCE, up to the commit "Update changelog for v1.10" 0f8c3875acac0350e513b5367bf1c10b7a58a4e9 .

This means that a pre-release version of Brogue Lite 1.10 [can be compiled from](https://github.com/HomebrewHomunculus/BrogueCE/actions/runs/1046494772) source using this repository. There are no releases in this repository yet, though, because the "Lite" features that I want to include to be "releaseworthy" are not all ready yet. 

Doing what is listed here will resolve the issue https://github.com/HomebrewHomunculus/BrogueLite/issues/30

## Planned changeset for next release

So what should be done before releasing a Brogue Lite 1.10?

#### Build pipeline chores
- [ ] Test build pipeline (GitHub Actions)
- [x] Merge upstream changes (will require resolving some conflicts) from [Brogue CE v1.10.1](https://github.com/tmewett/BrogueCE/releases/tag/v1.10.1) (up to eb3df24344329fa295996e523bf04c705722386e): #6
- [x] Make sure building still works after the merge
    * [ ] Fix MacOS build in GHA
- [x] Make sure all the resulting artifacts refer to "Lite" and not "CE"
- [x] Make [the changelog file](https://github.com/HomebrewHomunculus/BrogueCE/blob/master/CHANGELOG.md) into two separate files for CE and Lite, to avoid conflicts. #8 
- Maybe rename this repo to BrogueLite? Rename the old one to something else, and make sure to link to this one everywhere. Possibly archive the old repo? (Does GitHub show archived repos to the public or just the owners?)

#### New Lite features
- [ ] Finish making and testing gold/scoring changes
    * Original issues: https://github.com/HomebrewHomunculus/BrogueCE/pull/3 and https://github.com/HomebrewHomunculus/BrogueLite/issues/20
   * Note: all score-related changes should be done in a single release! So we shouldn't e.g. add depth scoring in one release and then tweak gold score amounts in the next release. It would be quite a bad player experience to have to adjust to several repeated scoring system changes. 
   * So, **if the testing and balancing of this becomes too big of a task, it can be dropped out of scope** for this release.
- [ ] Interface improvements
   * [x] Inventory display indicates stat requirements that you do not meet. Original issue: https://github.com/HomebrewHomunculus/BrogueLite/issues/16, PR: #12
   * [ ] If possible, make the inventory display more readable. Use colour coding of item category and/or kind. Or add subheaders. Original issue: https://github.com/HomebrewHomunculus/BrogueLite/issues/17
- [ ] Input improvements
    * [ ] Allow Angband-style diagonal movement keys. Original issue: https://github.com/HomebrewHomunculus/BrogueLite/issues/22

#### Already included
- [x] Migrate the repository into a properly forked one
- [x] Set up GitHub Actions builds
- [x] The player starts with 2 door keys. Door keys are fungible (one can be used for any door on any floor) and stack in the inventory display (though they still count towards the inventory limit).
   * Added in PRs https://github.com/HomebrewHomunculus/BrogueLite/pull/27 and https://github.com/HomebrewHomunculus/BrogueLite/pull/29 .
   * This has been tested and now works correctly on iron doors. However, other types of keys (orbs, cage keys and whatnot) have not been tested. 
   * [x] Rework easy mode into a testing mode. #9 PR: #13
   * [ ] Test orbs etc. to make sure that the key spawning/using code changes did not break them.

**Comment by HomebrewHomunculus** [comment]
Unfortunately, the Github Actions build broken at some point.

In any case, I've merged changes from BrogueCE v1.11.1.

I've also created a pre-release:
* https://github.com/HomebrewHomunculus/BrogueCE/releases/tag/Lite-1.11.1-RC1

I'm going to consider this issue closed and put the GHA fix on the backlog. For now, I've built Linux artifacts, and others can build their own, so I'll consider GHA buiilds non-critical.

These new releases contain the gameplay change that keys can be used on any door. However, they do not contain the scoring system rework mentioned above. I bit off a bit more than I could chew there, so I think that'll have to be a separate release.
* https://github.com/HomebrewHomunculus/BrogueCE/pull/3
